.github/workflows/go-build-and-publish.yml

```diff
@@ -0,0 +1,38 @@
+name: go-build-and-publish
+
+on:
+  push:
+    branches:
+      - main
+  release:
+    types:
+      - created
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.19.2
+
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Configure Agent
+        run: go run mage.go ConfigureAgent
+
+      - name: Test
+        run: mage Test
+
+      - name: Cross Compile
+        run: mage XBuildAll
+
+      - name: Publish
```

Comment:
Not sure we want to publish in this workflow.  

Also being that this is a scaffold repository, it may be useful to just build/test and code coverage.  There is a workflow we currently use for this in porter/operator that may fit here.  

I'll defer to @schristoff with what we want as a workflow in this repository.

Reply:
Original issue seemed to want same capabilities of ADO pipeline for GH - I think it's more useful for users building the scaffolding then anything. I'd say keep it (for now), I'm open to removing if it becomes a hassle.

Reply:
@schristoff @troy0820 what changes should be done here? Let me know.